apps/www/registry/default/example/toc-nested.tsx

```diff
@@ -0,0 +1,36 @@
+import {
+  TableOfContentsItem,
+  TableOfContentsLink,
+  TableOfContentsList,
+  TableOfContentsTitle,
+} from "../ui/toc"
+
+export default function TOCDNested() {
+  return (
+    <TableOfContentsList>
+      <TableOfContentsTitle>Nested Table of Contents</TableOfContentsTitle>
+      <TableOfContentsItem>
+        <TableOfContentsLink href="#first-section-nested">
```

Comment:
Can't we have the section id specified automatically?

Reply:
What do you mean by section id?
Can you give me an example of how you think this could be implemented?

Reply:
do you mean like this?

```ts

const TableOfContentsLink = React.forwardRef<
  React.ElementRef<"a">,
  TableOfContentsLinkProps
>(({ className, href, children, isActive, ...props }, ref) => {
  const linkTo =
    href || `#${children?.toString().toLowerCase().replace(/ /g, "-")}`
  return (
    <a
      ref={ref}
      className={cn(
        "text-sm font-medium text-foreground transition-colors hover:text-primary",
        isActive ? "font-medium text-foreground" : "text-muted-foreground",
        className
      )}
      href={linkTo}
      {...props}
    >
      {children}
    </a>
  )
})
TableOfContentsLink.displayName = "TableOfContentsLink"
```

Reply:
that's correct.